### DISCUSSION TITLE
[4.x]: No relationships found when using `releatedTo` via disabled matrix block containing an element field

### DISCUSSION THREAD

**Discussion by rob-c-baker** [craftcms/cms#13927] [Ideas]
### Description

This https://craftcms.com/docs/4.x/relations.html#relations-via-matrix describes relations via matrix blocks and how they are queried with the `[field].[sub-field]` type notation. In other places on that page it talks about generating queries to then pass in to a `relatedTo` parameter to customise the criteria applied to the relation.

I could, for example, do this _(borrowed from that page)_:

```twig
{% set alternateProteins = recipe.substitutions.foodGroup('protein').status('disabled').all() %}

{% set recipes = craft.entries()
  .section('recipes')
  .relatedTo({
    targetElement: alternateProteins,
    field: 'ingredients',
  })
  .all() %}
```

_Note the `.status('disabled')` condition._

However, I cannot do this with the "via Matrix" option as follows if the matrix blocks, referrered to as `steps.ingredient` being queried are in a disabled state, thus making the contained relations in the assets / entries / elements field invisible:

```twig
{% set relatedRecipes = craft.entries()
    .section('recipes')
    .relatedTo({
        targetElement: ingredient,
        field: 'steps.ingredient'
    })
    .all() %}
```

This is because of conditions hard-coded into `ElementRelationParamParser`:

https://github.com/craftcms/cms/blob/0fb938cba82b0ae9ee9e0854fd6bf96b428d6f2e/src/elements/db/ElementRelationParamParser.php#L401

https://github.com/craftcms/cms/blob/0fb938cba82b0ae9ee9e0854fd6bf96b428d6f2e/src/elements/db/ElementRelationParamParser.php#L428

### Expected behavior

Some way of adjusting the linked conditions - dropping those hard coded `enabled` conditions for example.

### Actual behavior

I have been able to access those conditions by listening to the `ElementQuery::EVENT_AFTER_PREPARE` event and picking through the `WHERE` conditions of the `subQuery` but it's very complex to find the right conditions to adjust in a generalised, safe way for now and the future.

This is the sort of stomach churning thing needed in the event listener:

```php
unset($event->sender->subQuery->where[5][2]['elements.id']->where[1]['source_matrixelements1.enabled']);
```

However, this is not the complete picture as when modifying a normalised query like this, I am very much aware I am in the "behaviour undefined" territory. The best I have been able to do with this approach so far is that it finds the first related asset via an Assets field on a disabled Matrix block but not the others.

### Craft CMS version

4.5.9

### PHP version

8.1

### Operating system and version

N/A

### Database type and version

N/A

### Image driver and version

N/A

### Installed plugins and versions

N/A

**Comment by brandonkelly** [comment]
This is working as intended right now, so I’ve converted your issue into a discussion.

We can look into making this possible when we get around to adding multi-step and bi-directional support to `relatedTo` (#8424).
